src/FsToolkit.ErrorHandling/AsyncValidationCE.fs

```diff
@@ -0,0 +1,158 @@
+﻿namespace FsToolkit.ErrorHandling
+
+open System
+
+[<AutoOpen>]
+module AsyncValidationCE =
+
+    type AsyncValidationBuilder() =
+        member inline _.Return(value: 'ok) : AsyncValidation<'ok, 'error> = AsyncValidation.ok value
+
+        member inline _.ReturnFrom
+            (result: AsyncValidation<'ok, 'error>)
+            : AsyncValidation<'ok, 'error> =
+            result
+
+        member inline _.Bind
+            (
+                result: AsyncValidation<'okInput, 'error>,
+                [<InlineIfLambda>] binder: 'okInput -> AsyncValidation<'okOutput, 'error>
+            ) : AsyncValidation<'okOutput, 'error> =
+            AsyncValidation.bind binder result
+
+        member inline this.Zero() : AsyncValidation<unit, 'error> = this.Return()
+
+        member inline _.Delay
+            ([<InlineIfLambda>] generator: unit -> AsyncValidation<'ok, 'error>)
+            : unit -> AsyncValidation<'ok, 'error> =
+            generator
+
+        member inline _.Run
+            ([<InlineIfLambda>] generator: unit -> AsyncValidation<'ok, 'error>)
+            : AsyncValidation<'ok, 'error> =
+            generator ()
+
+        member inline this.Combine
+            (
+                result: AsyncValidation<unit, 'error>,
+                [<InlineIfLambda>] binder: unit -> AsyncValidation<'ok, 'error>
+            ) : AsyncValidation<'ok, 'error> =
+            this.Bind(result, binder)
+
+        member inline this.TryWith
+            (
+                [<InlineIfLambda>] generator: unit -> AsyncValidation<'ok, 'error>,
+                [<InlineIfLambda>] handler: exn -> AsyncValidation<'ok, 'error>
+            ) : AsyncValidation<'ok, 'error> =
+            try
+                this.Run generator
+            with e ->
+                handler e
+
+        member inline this.TryFinally
+            (
+                [<InlineIfLambda>] generator: unit -> AsyncValidation<'ok, 'error>,
+                [<InlineIfLambda>] compensation: unit -> unit
+            ) : AsyncValidation<'ok, 'error> =
+            try
+                this.Run generator
+            finally
+                compensation ()
```

Comment:
This needs to be wrapped in an `async` or passed along to the `async.TryFinally` function.